packages/query/src/block-processor.ts

```diff
@@ -251,6 +252,26 @@ export class BlockProcessor implements BlockProcessorInterface {
       let metadata: Metadata | undefined;
       metadata = await this.querier.shieldedPool.assetMetadata(assetId);
 
+      // If the metadata is for a delegation token, customize its symbol.
+      if (metadata && assetPatterns.delegationTokenPattern.test(metadata.display)) {
+        // We can't trust the validator's self-described name, so use their validator ID.
+        // We know it's delegation_penumbravalid1... so use substrings:
+
+        // TODO: what's the best way to handle delegation tokens to unknown validators?
+
+        // Find the index of '1' in the string
+        const index = metadata.display.indexOf('1');
+        // Get the first N characters after the '1'
+        const id = metadata.display.substring(index + 1, index + 1 + 24);
+
+        metadata.symbol = 'Delegated UM (' + id + '...)';
+      }
+
+      // TODO: unbonding tokens?
+
+      // Note: the below code is incorrect, the asset ID is the hash of the denom,
+      // so this is actually generating metadata for a different asset. Not sure
+      // when/if this is used.
```

Comment:
This block of code seems to be redundant to: https://github.com/penumbra-zone/web/blob/main/packages/types/src/asset.ts which we should fix too. However, it appears to be the same as https://github.com/penumbra-zone/penumbra/blob/6ab217514f49c03eafbca1399fe57e19e6a96925/crates/core/asset/src/asset/id.rs#L97-L105. Or no?

Reply:
The difference is that the Display impl there isn't creating an entire Metadata, it's just converting to a string. It's correct to render the raw asset ID using bech32, the incorrect part is calling that string a "denom". But probably it doesn't cause problems if it's only ever used as a display data...